[BUG][Go] Missing imports for array of files and date-time parameters

#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When generating Go client code for an endpoint that accepts an array of binary files (e.g., `type: array, items: {type: string, format: binary}`), the generator fails to add the `import "os"` statement to the generated API file, even though the parameter type is `[]*os.File`. This results in a compilation error in the generated Go code.

The same issue affects array parameters with `format: date-time`, where the `import "time"` statement is missing.

##### openapi-generator version

7.18.0-SNAPSHOT (master branch as of November 19, 2025)

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  title: Test API
  version: 1.0.0
paths:
  '/upload':
    post:
      operationId: uploadFiles
      requestBody:
        required: true
        content:
          multipart/mixed:
            schema:
              type: object
              properties:
                files:
                  type: array
                  items:
                    type: string
                    format: binary
      responses:
        '200':
          description: Success
  '/events':
    get:
      operationId: getEvents
      parameters:
        - name: timestamps
          in: query
          schema:
            type: array
            items:
              type: string
              format: date-time
      responses:
        '200':
          description: Success
```

##### Generation Details

```bash
openapi-generator-cli generate \
    --additional-properties packageName=client \
    --additional-properties withGoCodegenComment=true \
    -i test-binary-array.yaml \
    -g go \
    -o ./generated
```

##### Steps to reproduce

1. Create an OpenAPI spec with an endpoint that has an array parameter with `format: binary` (see example above)
2. Generate Go client code using the `go` generator
3. Check the generated API file (e.g., `api_default.go`) - notice it's missing `import "os"`
4. Try to compile the generated code:
   ```bash
   cd generated
   go build
   ```
5. Compilation fails with error: 
```
./api_default.go:29:16: undefined: time
./api_default.go:32:54: undefined: time
./api_default.go:132:11: undefined: os
./api_default.go:135:47: undefined: os
```

##### Related issues/PRs

Issues containing `os.File`: 
https://github.com/OpenAPITools/openapi-generator/search?q=os.File&type=Issues

Open issues are very old, some might apply, but are very outdated. 

Similar issues may exist for other array-based type mappings

##### Suggest a fix

**Root Cause:**
In `AbstractGoCodegen.java`, the methods `postProcessOperationsWithModels()` and `postProcessWebhooksWithModels()` only check if `param.dataType` equals `"*os.File"` or `"time.Time"`. This works for single file/datetime parameters but fails for arrays.

For array parameters:
- Single file: `param.dataType = "*os.File"` ✓
- Array of files: `param.dataType = "[]*os.File"`, but `param.items.dataType = "*os.File"` ✗

**Solution:**
The import detection logic needs to also check `param.items.dataType` for array parameters, similar to how it's already correctly implemented in `postProcessModels()` for model properties.

**Required changes in `AbstractGoCodegen.java`:**

1. In `postProcessOperationsWithModels()` method (~line 550):
```java
// OLD CODE:
if (!addedOSImport && "*os.File".equals(param.dataType)) {
    imports.add(createMapping("import", "os"));
    addedOSImport = true;
}

if (!addedTimeImport && "time.Time".equals(param.dataType)) {
    imports.add(createMapping("import", "time"));
    addedTimeImport = true;
}

// NEW CODE:
if (!addedOSImport && ("*os.File".equals(param.dataType) ||
        (param.items != null && "*os.File".equals(param.items.dataType)))) {
    imports.add(createMapping("import", "os"));
    addedOSImport = true;
}

if (!addedTimeImport && ("time.Time".equals(param.dataType) ||
        (param.items != null && "time.Time".equals(param.items.dataType)))) {
    imports.add(createMapping("import", "time"));
    addedTimeImport = true;
}
```

2. In `postProcessWebhooksWithModels()` method (~line 651): Apply the same change

**Why this works:**
- For single parameters, `param.dataType` is checked directly
- For array parameters, `param.items` contains the element type information
- This aligns with the existing correct implementation in `postProcessModels()` (line ~780)

I will provide a PR with the fix

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java
@@ -547,13 +547,15 @@ public abstract class AbstractGoCodegen extends DefaultCodegen implements Codege
             }
             for (CodegenParameter param : operation.allParams) {
                 // import "os" if the operation uses files
-                if (!addedOSImport && "*os.File".equals(param.dataType)) {
+                if (!addedOSImport && ("*os.File".equals(param.dataType) ||
+                        (param.items != null && "*os.File".equals(param.items.dataType)))) {
                     imports.add(createMapping("import", "os"));
                     addedOSImport = true;
                 }
 
                 // import "time" if the operation has a time parameter.
-                if (!addedTimeImport && "time.Time".equals(param.dataType)) {
+                if (!addedTimeImport && ("time.Time".equals(param.dataType) ||
+                        (param.items != null && "time.Time".equals(param.items.dataType)))) {
                     imports.add(createMapping("import", "time"));
                     addedTimeImport = true;
                 }
@@ -648,13 +650,15 @@ public abstract class AbstractGoCodegen extends DefaultCodegen implements Codege
             }
             for (CodegenParameter param : operation.allParams) {
                 // import "os" if the operation uses files
-                if (!addedOSImport && "*os.File".equals(param.dataType)) {
+                if (!addedOSImport && ("*os.File".equals(param.dataType) ||
+                        (param.items != null && "*os.File".equals(param.items.dataType)))) {
                     imports.add(createMapping("import", "os"));
                     addedOSImport = true;
                 }
 
                 // import "time" if the operation has a time parameter.
-                if (!addedTimeImport && "time.Time".equals(param.dataType)) {
+                if (!addedTimeImport && ("time.Time".equals(param.dataType) ||
+                        (param.items != null && "time.Time".equals(param.items.dataType)))) {
                     imports.add(createMapping("import", "time"));
                     addedTimeImport = true;
                 }

--- a/modules/openapi-generator/src/test/java/org/openapitools/codegen/go/AbstractGoCodegenTest.java
+++ b/modules/openapi-generator/src/test/java/org/openapitools/codegen/go/AbstractGoCodegenTest.java
@@ -19,14 +19,26 @@ package org.openapitools.codegen.go;
 
 import io.swagger.v3.oas.models.Components;
 import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
 import io.swagger.v3.oas.models.media.*;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.parameters.RequestBody;
 import org.mockito.Answers;
+import org.openapitools.codegen.*;
 import org.openapitools.codegen.CodegenConstants;
 import org.openapitools.codegen.languages.AbstractGoCodegen;
+import org.openapitools.codegen.model.OperationMap;
+import org.openapitools.codegen.model.OperationsMap;
+import org.openapitools.codegen.model.WebhooksMap;
 import org.openapitools.codegen.utils.ModelUtils;
 import org.testng.Assert;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
 
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.withSettings;
@@ -107,5 +119,92 @@ public class AbstractGoCodegenTest {
         ModelUtils.setGenerateAliasAsModel(false);
         defaultValue = codegen.getTypeDeclaration(schema);
         Assert.assertEquals(defaultValue, "map[string]interface{}");
+    }
+
+    @Test(description = "test that os import is added for array of binary parameters in operations")
+    public void testOsImportForArrayOfBinaryParametersInOperations() {
+        // Create OpenAPI spec with array of binary files
+        OpenAPI openAPI = new OpenAPI();
+        openAPI.setComponents(new Components());
+        
+        // Create operation with array of binary parameter in request body
+        Operation operation = new Operation();
+        RequestBody requestBody = new RequestBody();
+        Content content = new Content();
+        MediaType mediaType = new MediaType();
+        
+        Schema<?> arraySchema = new ArraySchema().items(
+            new BinarySchema()
+        );
+        
+        ObjectSchema objectSchema = new ObjectSchema();
+        objectSchema.addProperty("files", arraySchema);
+        mediaType.setSchema(objectSchema);
+        content.addMediaType("multipart/form-data", mediaType);
+        requestBody.setContent(content);
+        operation.setRequestBody(requestBody);
+        
+        codegen.setOpenAPI(openAPI);
+        
+        // Convert to CodegenOperation
+        CodegenOperation codegenOperation = codegen.fromOperation("/upload", "post", operation, null);
+        
+        // Create OperationsMap structure
+        OperationMap operationMap = new OperationMap();
+        operationMap.setOperation(codegenOperation);
+        OperationsMap operationsMap = new OperationsMap();
+        operationsMap.setOperation(operationMap);
+        operationsMap.setImports(new ArrayList<>());
+        
+        // Post-process the operations
+        OperationsMap result = codegen.postProcessOperationsWithModels(operationsMap, Collections.emptyList());
+        
+        // Assert that "os" import was added
+        List<Map<String, String>> imports = result.getImports();
+        boolean hasOsImport = imports.stream()
+            .anyMatch(imp -> "os".equals(imp.get("import")));
+        
+        Assert.assertTrue(hasOsImport, "Expected 'os' import to be added for array of binary files");
+    }
+    
+    @Test(description = "test that time import is added for array of date-time parameters in operations")
+    public void testTimeImportForArrayOfDateTimeParametersInOperations() {
+        // Create OpenAPI spec with array of date-time parameter
+        OpenAPI openAPI = new OpenAPI();
+        openAPI.setComponents(new Components());
+        
+        Operation operation = new Operation();
+        Parameter parameter = new Parameter();
+        parameter.setName("timestamps");
+        parameter.setIn("query");
+        
+        ArraySchema arraySchema = new ArraySchema();
+        DateTimeSchema dateTimeSchema = new DateTimeSchema();
+        arraySchema.setItems(dateTimeSchema);
+        parameter.setSchema(arraySchema);
+        
+        operation.addParametersItem(parameter);
+        
+        codegen.setOpenAPI(openAPI);
+        
+        // Convert to CodegenOperation
+        CodegenOperation codegenOperation = codegen.fromOperation("/events", "get", operation, null);
+        
+        // Create OperationsMap structure
+        OperationMap operationMap = new OperationMap();
+        operationMap.setOperation(codegenOperation);
+        OperationsMap operationsMap = new OperationsMap();
+        operationsMap.setOperation(operationMap);
+        operationsMap.setImports(new ArrayList<>());
+        
+        // Post-process the operations
+        OperationsMap result = codegen.postProcessOperationsWithModels(operationsMap, Collections.emptyList());
+        
+        // Assert that "time" import was added
+        List<Map<String, String>> imports = result.getImports();
+        boolean hasTimeImport = imports.stream()
+            .anyMatch(imp -> "time".equals(imp.get("import")));
+        
+        Assert.assertTrue(hasTimeImport, "Expected 'time' import to be added for array of date-time parameters");
     }
 }

